packages/abi-utils/lib/arbitrary.ts

```diff
@@ -393,29 +392,30 @@ const reservedWords = new Set([
   "years"
 ]);
 
-// borrowed from https://runkit.com/dubzzz/faker-to-fast-check
-const fakerToArb = (template: string, transform = camelCase) => {
-  return fc
-    .integer()
-    .noBias()
-    .noShrink()
-    .map(seed => {
-      faker.seed(seed);
-      return transform(faker.fake(template));
-    })
-    .filter(word => !reservedWords.has(word));
+type WordListKey = keyof typeof wordLists;
+const Name = (
+  wordTypes: WordListKey[] = ["noun"],
+  transform = camelCase
+): fc.Arbitrary<string> => {
+  const wordArbitraries = wordTypes.map(wordType =>
+    fc.constantFrom(...wordLists[wordType])
+  );
+
+  const wordsArbitrary = fc.tuple(...wordArbitraries);
+
+  const nameArbitrary = wordsArbitrary.map(words => transform(words.join(" ")));
+
+  return nameArbitrary.filter(word => !reservedWords.has(word));
 };
 
 const ParameterName = () =>
   fc.oneof(
-    { arbitrary: fakerToArb("{{hacker.noun}}"), weight: 9 },
+    { arbitrary: Name(["noun"]), weight: 9 },
     { arbitrary: fc.constant(""), weight: 1 }
   );
-const EventName = () =>
-  fakerToArb("{{hacker.verb}} {{hacker.noun}}", pascalCase);
-const ErrorName = () =>
-  fakerToArb("{{hacker.noun}} {{hacker.noun}}", pascalCase);
-const FunctionName = () => fakerToArb("{{hacker.verb}} {{hacker.noun}}");
+const EventName = () => Name(["verb", "noun"], pascalCase);
+const ErrorName = () => Name(["noun", "noun"], pascalCase);
+const FunctionName = () => Name(["verb", "noun"]);
```

Comment:
I really like how this makes it seem like fast-check just wants to play Mad Libs